packages/compass-query-bar/src/components/query-history/index.spec.tsx

```diff
@@ -0,0 +1,245 @@
+import React from 'react';
+import { applyMiddleware, createStore as _createStore } from 'redux';
+import thunk from 'redux-thunk';
+import { expect } from 'chai';
+import {
+  render,
+  screen,
+  waitForElementToBeRemoved,
+  within,
+} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { Provider } from 'react-redux';
+import Sinon from 'sinon';
+import fs from 'fs';
+import os from 'os';
+
+import QueryHistory from '.';
+import { FavoriteQueryStorage, RecentQueryStorage } from '../../utils';
+import {
+  queryBarReducer,
+  INITIAL_STATE,
+  fetchRecents,
+  fetchFavorites,
+} from '../../stores/query-bar-reducer';
+
+const BASE_QUERY = {
+  filter: { name: 'hello' },
+  project: { name: 1 },
+  sort: { name: -1 },
+  collation: { locale: 'en' },
+  skip: 10,
+  limit: 20,
+};
+const RECENT_QUERY = {
+  _id: 'one',
+  _lastExecuted: new Date(),
+  ...BASE_QUERY,
+};
+
+const FAVORITE_QUERY = {
+  _id: 'two',
+  _name: 'Best query',
+  _lastExecuted: new Date(),
+  ...BASE_QUERY,
+};
+
+function createStore(basepath?: string) {
+  const favoriteQueryStorage = new FavoriteQueryStorage(basepath);
+  const recentQueryStorage = new RecentQueryStorage(basepath);
+
+  const store = _createStore(
+    queryBarReducer,
+    {
+      ...INITIAL_STATE,
+      namespace: 'airbnb.listings',
+      host: 'localhost',
+    } as any,
+    applyMiddleware(
+      thunk.withExtraArgument({
+        favoriteQueryStorage,
+        recentQueryStorage,
+      })
+    )
+  );
+
+  return {
+    store,
+    favoriteQueryStorage,
+    recentQueryStorage,
+  };
+}
+
+const renderQueryHistory = (basepath?: string) => {
+  const data = createStore(basepath);
+  render(
+    <Provider store={data.store}>
+      <QueryHistory />
+    </Provider>
+  );
+  return data;
+};
+
+describe('query-history', function () {
+  let tmpDir: string;
+  before(function () {
+    tmpDir = fs.mkdtempSync(os.tmpdir());
+  });
+
+  after(function () {
+    fs.rmdirSync(tmpDir, { recursive: true });
+  });
+
+  context('zero state', function () {
+    it('in recents', function () {
+      renderQueryHistory();
+      userEvent.click(screen.getByText(/recents/i));
+      expect(
+        screen.getByText(/your recent queries will appear here\./i)
+      ).to.exist;
+    });
+
+    it('in favorites', function () {
+      renderQueryHistory();
+      userEvent.click(screen.getByText(/favorites/i));
+      expect(
+        screen.getByText(/your favorite queries will appear here\./i)
+      ).to.exist;
+    });
```

Comment:
only one UI change (34d6938349a885180d20338d0eb8f17c712e6832) i did in this PR - to render different text for different tabs in Zero state. If its not ok, I'll revert it.